### PR TITLE
Unroll FeZero, FeOne

### PR DESCRIPTION
**Pull request by ValarDragon**
This unrolls FeZero and FeOne. This doesn't change the output of the functions, but increases its speed:
```
$ benchcmp old.txt new.txt
benchmark                    old ns/op     new ns/op     delta
BenchmarkKeyGeneration-8     100659        99175         -1.47%
BenchmarkSigning-8           102219        100941        -1.25%
BenchmarkVerification-8      278957        275965        -1.07%
BenchmarkKeyGeneration-8     120965        119236        -1.43%
BenchmarkMap-8               36722         36478         -0.66%
```

The idea was taken from https://github.com/agl/ed25519/pull/11

### CODE REVIEW

**Review by liamsi — APPROVED** [review]
LGTM! 

Not sure if we'd want to deviate further from AGL's (or golang/x/crypto's) code for these minor performance improvements though.

I'll leave it to the admins of this repo to decide (I don't have write access here anyways).

**Comment by ValarDragon** [comment]
@ebuchman can this be merged?

**Comment by ValarDragon** [comment]
repo abandoned.

**Comment by cwgoes** [comment]
Does Golang's compiler really not do this?

(evidently not, per benchmarks)

**Comment by ValarDragon** [comment]
Theres a lot of fighting the compiler you have to do for performant golang code :(, the compiler doesn't do many optimizations for you. (See https://www.youtube.com/watch?v=keydVd-Zn80 for the struggles of in-lining)

**Pull request closed without merging**
